[doc] Fix linalg.slogdet doc consistency issues

heitorschueroff opened the pull request:
This PR is part of a larger effort to ensure torch.linalg documentation is consistent (see https://github.com/pytorch/pytorch/issues/50287).

Updated torch.linalg.slogdet to include notes about cross-device synchronization, backend routines used and fix signature missing out argument.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51301 [doc] Add deprecation message to torch.slogdet in favor of torch.linalg.slogdet
* **#51300 [doc] Fix linalg.slogdet doc consistency issues**

Differential Revision: [D26147898](https://our.internmc.facebook.com/intern/diff/D26147898)

heitorschueroff committed:
[doc] Fix linalg.slogdet doc consistency issues

[ghstack-poisoned]

mruberry commented on torch/linalg/__init__.py:
```diff
@@ -148,23 +148,30 @@
 Supports input of float, double, cfloat and cdouble datatypes.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU
```
I wouldn't add "while checking if the call to MAGMA was successful using the info argument." because "info" is an implementation detail the reader may not be familiar with. Also the MAGMA algorithm may run on CPU.

mruberry commented on torch/linalg/__init__.py:
```diff
@@ -148,23 +148,30 @@
+.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU
+          while checking if the call to MAGMA was successful using the info argument.
+
+.. note:: The determinant is computed via LU factorization. On CPU, this is done using the LAPACK
```
"via" -> "using"

"On CPU, this is done using the LAPACK..." -> "LAPACK's \`getrf\` is used for CPU inputs, and MAGMA's \`getrf\` is used for CUDA inputs."

mruberry reviewed (approved):
Just a couple tweaks, otherwise looks great!

heitorschueroff committed:
Update on "[doc] Fix linalg.slogdet doc consistency issues"

This PR is part of a larger effort to ensure torch.linalg documentation is consistent (see https://github.com/pytorch/pytorch/issues/50287).

Updated torch.linalg.slogdet to include notes about cross-device synchronization, backend routines used and fix signature missing out argument.




[ghstack-poisoned]
